### PR TITLE
Issue fixing while flow testing

### PR DESCRIPTION
**Pull request by kanika1894**

### CODE REVIEW

File: src/configure/helper/mustacheHelper.ts [hunk 1]
```diff
@@ -220,12 +220,14 @@ export class MustacheHelper {
         }
 
         var result: Object = {};
-        Object.keys(mustacheObject).forEach(key => {
-            if (!!key && !!mustacheObject[key]) {
-                result[key] = MustacheHelper.renderObject(mustacheObject[key], view);
-            }
-        });
+        if (mustacheObject) {
```
Comment (anuragc617) [inline src/configure/helper/mustacheHelper.ts:223]: why this condition?

File: src/configure/templateInputHelper/InputControlProvider.ts [hunk 2]
```diff
@@ -60,9 +60,12 @@ export class InputControlProvider {
             let dependentInputControlArray = this._getInputDependencyArray(inputControl, [properties[element]], false);
             let dependentClientInputMap = this._getClientDependencyMap(inputControl, [properties[element]]);
             let newValue = this._computeMustacheValue(properties[element], dependentInputControlArray, dependentClientInputMap);
+            if (!Object.keys(newValue).length) {
+                newValue = properties[element];
+            }
             inputControl.updateInputDescriptorProperty(key, newValue);
             if (key === constants.inputModeProperty) {
-                let updatedControlType = InputControlUtility.getInputControlType(InputMode[newValue]);
+                let updatedControlType = InputControlUtility.getInputControlType(parseInt(newValue.toString()));
```
Comment (anuragc617) [inline src/configure/templateInputHelper/InputControlProvider.ts:68]: why this change?

File: src/configure/templateInputHelper/InputControlProvider.ts [hunk 3]
```diff
@@ -148,12 +151,11 @@ export class InputControlProvider {
     }
 
     private _computeMustacheValue(mustacheExpression: string, dependentInputControlArray: InputControl[], dependentClientInputMap: StringMap<any>): string {
-
         var dependentInputValues = this._getInputParameterValueIfAllSet(dependentInputControlArray);
         if (dependentInputControlArray && dependentInputControlArray.length > 0 && !dependentInputValues) {
             return "";
         } else {
-            return MustacheHelper.render(mustacheExpression, { inputs: dependentInputValues, client: dependentClientInputMap });
+            return MustacheHelper.renderObject(mustacheExpression, { inputs: dependentInputValues, client: dependentClientInputMap });
```
Comment (anuragc617) [inline src/configure/templateInputHelper/InputControlProvider.ts:158]: which case is not working with render()?

File: src/configure/configure.ts [hunk 4]
```diff
@@ -171,6 +171,7 @@ class Orchestrator {
                 else {
                     this.context['rightClickScenario'] = true;
                 }
+                this.context['resourceId'] = this.inputs.targetResource.resource.id;
```
Comment (anuragc617) [inline src/configure/configure.ts:174]: It will fail for non-webApp case ex AKS, because resource will be undefined there and it will throw an exception. 
Should we consider rightClickScenario=true for WebApp ??

File: src/configure/templateInputHelper/InputControlProvider.ts (hunk 3 above)
Comment (kanika1894) [reply src/configure/templateInputHelper/InputControlProvider.ts:158]: Just because renderObject is also handling the string scenario and calling render for the same, so updated it just for improvement purpose.

File: src/configure/templateInputHelper/InputControlProvider.ts (hunk 2 above)
Comment (kanika1894) [reply src/configure/templateInputHelper/InputControlProvider.ts:68]: The InputMode in template can be an integer or string. Example, if it's a override scenario, it will be treated as a string and direct inputMode property comes as a number. To handle both the scenarios, this is done.

File: src/configure/helper/mustacheHelper.ts (hunk 1 above)
Comment (kanika1894) [reply src/configure/helper/mustacheHelper.ts:223]: It throws a null exception if null is passed as a mustacheObject.
Example scenario : changing dataSourceId of a parameter to null in overrides property.

File: src/configure/configure.ts (hunk 4 above)
Comment (kanika1894) [reply src/configure/configure.ts:174]: ``
  private async getAzureResource(targetType: TargetResourceType) {
        var azureResourceSelector = ResourceSelectorFactory.getAzureResourceSelector(targetType);
        this.inputs.targetResource.resource = await azureResourceSelector.getAzureResource(this.inputs);
    }
``

We are using azureResourceSelector to get the resourceId. In case of AKS, it will be null because the flow is different for now., but it doesn't seem this will cause a problem?

File: src/configure/templateInputHelper/InputControlProvider.ts [hunk 5]
```diff
@@ -60,9 +60,12 @@ export class InputControlProvider {
             let dependentInputControlArray = this._getInputDependencyArray(inputControl, [properties[element]], false);
             let dependentClientInputMap = this._getClientDependencyMap(inputControl, [properties[element]]);
             let newValue = this._computeMustacheValue(properties[element], dependentInputControlArray, dependentClientInputMap);
+            if (!Object.keys(newValue).length) {
```
Comment (anuragc617) [inline src/configure/templateInputHelper/InputControlProvider.ts:63]: Why this condition is added?
